consensus/src/consensus.rs

```diff
@@ -186,7 +184,7 @@ impl<T: Operations + 'static, D: Database + 'static> Consensus<T, D> {
                 ru.base_timeouts.clone(),
             );
 
-            while iter < CONSENSUS_MAX_ITER {
```

Comment:
I agree. When reaching last iteration we'll want to exit the loop and wait for the Emergency Block

Reply:
> I agree. When reaching last iteration we'll want to exit the loop and wait for the Emergency Block

I disagree. When we'll implement the emergency block, we should NOT EXIT the loop when the last iteration is reached.
We should still wait for relaxed iterations messages and, in the meanwhile, we should request the emergency block.

Reply:
By exiting the loop I didn't mean terminate the execution. 
I meant stop running this loop and handle the situation accordingly (send the EB request, and listen for all messages that could end this round).
I simply dislike having the loop (which is intended to execute steps) get "stuck" on the last iteration just to wait on stuff.

Reply:
I believe we are going out of scope.
We should not take into consideration EmergencyBlock while reviewing this PR.

My original RFC was simply about get back to original while and panic after (just to notice it in case something goes wrong while testing)
```diff
-            while iter < CONSENSUS_MAX_ITER {
+            loop {
```

Said that, I believe that this PR is properly addressing the target issue